MichaelGalo/Impactree_API
# Added images property to charity & image support to CharityViewSet

Adding image support to Charity model to allow storing and retrieving charity logos and branding images.

## Changes

1. Added ImageField to Charity model with upload_to="charityimages"
2. Added media settings configuration in settings.py and urls.py
3. Updated CharitySerializer to include image field
4. Updated CharityViewSet to handle base64 image uploads
5. Added image testing support with sample base64 image

## Requests / Responses

**Request**

POST /charities Creates a new charity with image

```json
{
"name": "Example Charity",
"description": "A charity description",
"impact_metric": "Lives Saved",
"impact_ratio": 0.95,
"website_url": "https://example.com/",
"category": 1,
"image": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1..."
}
```

**Response**

HTTP/1.1 201 Created

```json
{
"id": 1,
"name": "Example Charity",
"description": "A charity description",
"impact_metric": "Lives Saved",
"impact_ratio": 0.95,
"website_url": "https://example.com/",
"category": {
"id": 1,
"name": "Category Name"
},
"image": "/media/charityimages/charity-uuid.png"
}
```

## Testing

1.  Run migrations to add image field to Charity model
2.  Run test suite to verify image upload functionality
3.  Test image upload through admin interface
4.  Verify image retrieval through API endpoints
5.  Check that only admin users can upload/modify images
